store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReaderBuilder.java

```diff
@@ -131,21 +203,66 @@ public CarbonReaderBuilder withBatch(int batch) {
   public CarbonReaderBuilder withHadoopConf(String key, String value) {
     if (this.hadoopConf == null) {
       this.hadoopConf = new Configuration();
-
     }
     this.hadoopConf.set(key, value);
     return this;
   }
 
   /**
    * Configure Row Record Reader for reading.
-   *
    */
   public CarbonReaderBuilder withRowRecordReader() {
     this.useVectorReader = false;
     return this;
   }
 
+  public InputSplit[] getSplits() throws IOException {
+    if (hadoopConf == null) {
+      hadoopConf = FileFactory.getConfiguration();
+    }
+    CarbonTable table;
+    // now always infer schema. TODO:Refactor in next version.
+    table = CarbonTable.buildTable(tablePath, tableName, hadoopConf, false);
+    final CarbonFileInputFormat format = new CarbonFileInputFormat();
+    final Job job = new Job(hadoopConf);
+    Map<String, String> tableProperties = table.getTableInfo().getFactTable().getTableProperties();
+    tableProperties.put(CarbonCommonConstants.CACHE_LEVEL, "BLOCKLET");
```

Comment:
better to use CarbonCommonConstants.CACHE_LEVEL_DEFAULT_VALUE

Reply:
no need this code, removed